### PR TITLE
Introduced ability to set up scroll's type selector

### PR DESCRIPTION
**Pull request by bytenik**
Replaces #1685 which was on a branch that DNE anymore.

### CODE REVIEW

File: src/Nest/ElasticClient-Scroll.cs [hunk 1]
```diff
@@ -56,7 +58,9 @@ public partial class ElasticClient
 				{
 					string scrollId = p.ScrollId;
 					p.ScrollId = null;
-					return this.RawDispatch.ScrollDispatchAsync<SearchResponse<T>>(p, scrollId);
+				    p.DeserializationState(CreateScrollDeserializer<T, T>(d));
+
+                    return this.RawDispatch.ScrollDispatchAsync<SearchResponse<T>>(p, scrollId);
```
Comment (Mpdreamz) [inline src/Nest/ElasticClient-Scroll.cs]: This should go on the other overloads as well. Can you also make sure your PR uses tabs not spaces?

**Comment by Mpdreamz** [comment]
Appreciate the retargeting of the PR, much appreciated. I left some comments on the PR itself.

**Comment by bytenik** [comment]
Fixed.

**Comment by Mpdreamz** [comment]
ty!, I'll post back when this appears on myget. We are close to a 2.0.0-alpha release so the 1.x bleeding edge builds will probably move over to a separate myget feed.

**Comment by Mpdreamz** [comment]
@bytenik I know this was a blocker for you: we now have a separate myget feed for building the tip of the `1.x` branch continuously:

https://www.myget.org/feed/Packages/elasticsearch-net-legacy

Your changes should be in there!

**Comment by bytenik** [comment]
Looks like security settings are private for that repository. I can't hit
that URL.

**Comment by Mpdreamz** [comment]
If that one doesn't this one should: https://www.myget.org/gallery/elasticsearch-net-legacy

Although It might take 15 minutes to propagate according to myget

**Comment by issam-chouchane** [comment]
And how to us it in scroll request ??
what is the first and the second parms (poor docs ;()

`scrollRequest.TypeSelector(.......`

**Comment by Mpdreamz** [comment]
The docs actually mention this, we are getting there :smile: 

https://www.elastic.co/guide/en/elasticsearch/client/net-api/current/covariant-search-results.html#_using_covarianttypes

**Comment by issam-chouchane** [comment]
Good, i followed the steps as the doc said : 

Ibuild my scroll request first  :
```
private IScrollRequest GetScrollRequest(LogSearchRequest request)
        {
            ScrollRequest scrollRequest = new ScrollRequest("", request.Delay);
            if (request.RequestedTypes != null && request.RequestedTypes.Any())
            {
                List<TypeName> typeNames = new List<TypeName>();
                foreach (var type in request.RequestedTypes)
                {
                    TypeName typeName = new TypeName();
                    typeName.Name = type.Name;
                    typeName.Type = type;

                    typeNames.Add(typeName);
                }

                scrollRequest.CovariantTypes = Types.Type(typeNames);
            }
            return scrollRequest as IScrollRequest;
        }
```

After i use the output result in the scroll function :

```
...................ISearchRequest> request, IScrollRequest scrollRequest) where T : Entity
        {
            var elasticResponse = _elasticLogger.NestClient.Search<dynamic>(request);
            List<IHit<dynamic>> hits = new List<IHit<dynamic>>();
            scrollRequest.ScrollId = elasticResponse.ScrollId;

            if (elasticResponse.Total > 0)
            {
                do
                {
                    hits.AddRange(elasticResponse.Hits);
                    elasticResponse = _elasticLogger.NestClient.Scroll<dynamic>(scrollRequest);
                } while (elasticResponse.IsValid && elasticResponse.Documents.Any());
            } 
```

the first search work well and build a correct URL (cheking by fiddler), but in the scroll URL i got my Post object but the url doesn't contains the types, it look like "...../_search/scroll" with no types.

**Comment by issam-chouchane** [comment]
I used Search function to continue having my result resolved and deserialized as expected : 

```
            var elasticResponse = _elasticLogger.NestClient.Search<dynamic>(request);
            List<IHit<dynamic>> hits = new List<IHit<dynamic>>();

            if (elasticResponse.Total > 0)
            {
                do
                {
                    request.From += 10;
                    hits.AddRange(elasticResponse.Hits);
                    elasticResponse = _elasticLogger.NestClient.Search<dynamic>(request);
                } while (elasticResponse.IsValid && elasticResponse.Documents.Any());
            }
```

the scroll function still returns objects not deserialized.

**Pull request merged**